reactive/coredns.py

```diff
@@ -1,22 +1,57 @@
+from pathlib import Path
+
 from charmhelpers.core import hookenv
-from charms.reactive import set_flag, clear_flag
+from charms.reactive import set_flag, clear_flag, endpoint_from_flag
 from charms.reactive import when, when_not
 
 from charms import layer
 
 
+@when_not('cluster-dns.joined')
+def wait_for_cluster_dns():
+    layer.status.maintenance('Waiting for cluster dns relation')
```

Comment:
This should be a `blocked` status, not `maintenance`. See https://discourse.jujucharms.com/t/charm-unit-status-and-their-meanings/1168